### PR TITLE
fix: check if results are empty

### PR DESCRIPTION
**Pull request by alaeddine-13**
Fixing encoding of a dataset using cas.
To reproduce:
download this file https://jinaai.slack.com/files/U026WRMRL6A/F04GE6JKX0U/dataset.csv

then:
```
from docarray import DocumentArray, Document
with open('dataset.csv') as fp:
  da = DocumentArray.from_csv(fp)

from clip_client import Client

c = Client(
    'grpcs://api.clip.jina.ai:2096', credential={'Authorization': 'atuh-token'}
)

encoded_da = c.encode(da, show_progress=True)
```

### CODE REVIEW

**Comment by alaeddine-13** [comment]
now a server side issue happens:
```
    raise _create_rpc_error(await self.initial_metadata(), await
grpc.aio._call.AioRpcError: <AioRpcError of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "HTTP Balancer service in fail-fast |Gateway: Communication error with deployment clip_t at address(es) {'clip-t.grpc-flow.svc:8080'}. Head or worker(s) may be down."
	debug_error_string = "{"created":"@1671719158.543427000","description":"Error received from peer ipv4:52.58.81.82:2096","file":"src/core/lib/surface/call.cc","file_line":967,"grpc_message":"HTTP Balancer service in fail-fast |Gateway: Communication error with deployment clip_t at address(es) {'clip-t.grpc-flow.svc:8080'}. Head or worker(s) may be down.","grpc_status":14}"
```

File: client/clip_client/client.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def _gather_result(
         response, results: 'DocumentArray', attribute: Optional[str] = None
     ):
         r = response.data.docs
-        if attribute:
+        if attribute and len(r) > 0:
```
Comment (ZiniuYu) [inline client/clip_client/client.py:168]: Ideally `len(r)>0` must hold

Comment (jemmyshin) [inline client/clip_client/client.py:168]: why remove this line?

**Review by jemmyshin — CHANGES_REQUESTED** [review]

**Comment by ZiniuYu** [comment]
closing it since it's a connection issue

**Pull request closed without merging**